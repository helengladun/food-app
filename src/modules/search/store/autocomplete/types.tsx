export enum AutocompleteTypes {
    AUTOCOMPLETE = '@@search/AUTOCOMPLETE',
    AUTOCOMPLETE_SUCCESS = '@@search/AUTOCOMPLETE_SUCCESS',
    AUTOCOMPLETE_FAILURE = '@@search/AUTOCOMPLETE_FAILURE'
}

class AutocompleteAction {
    readonly type = AutocompleteTypes.AUTOCOMPLETE;
    constructor(public payload: {isLoading: boolean}) {};
}

class AutocompleteSuccessAction {
    readonly type = AutocompleteTypes.AUTOCOMPLETE_SUCCESS;
    constructor(public payload: {suggestions: string[], isLoading: boolean }) {}
}

class AutocompleteFailureAction {
    readonly type = AutocompleteTypes.AUTOCOMPLETE_FAILURE;
    constructor(public payload: {isLoading: boolean}, public error: []) {}
}

export type AutocompleteActions = AutocompleteAction | AutocompleteSuccessAction | AutocompleteFailureAction;
