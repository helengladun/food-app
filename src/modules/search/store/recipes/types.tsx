export enum GetRecipesTypes {
    GET_RECIPES = '@@search/GET_RECIPES',
    GET_RECIPES_SUCCESS = '@@search/GET_RECIPES_SUCCESS',
    GET_RECIPES_FAILURE = '@@search/GET_RECIPES_FAILURE',
}

class GetRecipesAction {
    readonly type = GetRecipesTypes.GET_RECIPES;
    constructor(public payload: {isLoading: boolean}) {};
}

class GetRecipesSuccessAction {
    readonly type = GetRecipesTypes.GET_RECIPES_SUCCESS;
    constructor(public payload: {recipes: string[], isLoading: boolean }) {}
}

class GetRecipesFailureAction {
    readonly type = GetRecipesTypes.GET_RECIPES_FAILURE;
    constructor(public payload: {isLoading: boolean}, public error: []) {}
}

export type GetRecipesActions = GetRecipesAction | GetRecipesSuccessAction | GetRecipesFailureAction;
