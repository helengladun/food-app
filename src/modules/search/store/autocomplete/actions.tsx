import { ActionCreator } from 'redux';
import { AutocompleteTypes, AutocompleteActions } from './types';

export const autocomplete:ActionCreator<AutocompleteActions>  = () => ({
    type: AutocompleteTypes.AUTOCOMPLETE,
    payload: {
        isLoading: true
    }
});

export const autocompleteSuccess: ActionCreator<AutocompleteActions> = (value: string[]) => ({
    type: AutocompleteTypes.AUTOCOMPLETE_SUCCESS,
    payload: {
        suggestions: value,
        isLoading: false
    }
});

export const autocompleteFailure: ActionCreator<AutocompleteActions> = (errors: []) => ({
    type: AutocompleteTypes.AUTOCOMPLETE_FAILURE,
    payload: {
        isLoading: false,
    },
    error: errors
});

