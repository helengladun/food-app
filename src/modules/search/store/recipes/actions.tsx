import { ActionCreator } from 'redux';
import { GetRecipesActions, GetRecipesTypes } from './types';

export const getRecipes:ActionCreator<GetRecipesActions>  = () => ({
    type: GetRecipesTypes.GET_RECIPES,
    payload: {
        isLoading: true
    }
});

export const getRecipesSuccess: ActionCreator<GetRecipesActions> = (value: string[]) => ({
    type: GetRecipesTypes.GET_RECIPES_SUCCESS,
    payload: {
        recipes: value,
        isLoading: false
    }
});

export const getRecipesFailure: ActionCreator<GetRecipesActions> = (errors: []) => ({
    type: GetRecipesTypes.GET_RECIPES_FAILURE,
    payload: {
        isLoading: false,
    },
    error: errors
});

