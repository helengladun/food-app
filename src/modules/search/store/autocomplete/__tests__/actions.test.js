import configureStore from 'redux-mock-store';
import * as autocompleteActions from '../actions';
import {AutocompleteTypes} from "../types";

const mockStore = configureStore();
const store = mockStore();

describe('autocomplete actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  describe('autocomplete start', () => {
    test('Dispatches the autocomplete action', () => {
      const expectedResult = [
        {
          payload: {
            isLoading: true
          },
          type: AutocompleteTypes.AUTOCOMPLETE
        }
      ];

      store.dispatch(autocompleteActions.autocomplete());
      expect(store.getActions()).toEqual(expectedResult);
    })
  });

  describe('autocomplete success', () => {
    test('Dispatches the autocomplete success action', () => {
      // const expectedResult = [
      //   {
      //     payload: {
      //       isLoading: true
      //     },
      //     type: AutocompleteTypes.AUTOCOMPLETE
      //   }
      // ];

      store.dispatch(autocompleteActions.autocompleteSuccess());
      // expect(store.getActions()).toEqual(expectedResult);
      expect(store.getActions()).toMatchSnapshot();
    })
  });

  describe('autocomplete failed', () => {
    test('Dispatches the autocomplete failed action', () => {
      store.dispatch(autocompleteActions.autocompleteFailure());
      expect(store.getActions()).toMatchSnapshot();
    })
  });

});
