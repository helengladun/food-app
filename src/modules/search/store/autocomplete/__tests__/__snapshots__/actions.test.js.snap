// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`autocomplete actions autocomplete failed Dispatches the autocomplete failed action 1`] = `
Array [
  Object {
    "error": undefined,
    "payload": Object {
      "isLoading": false,
    },
    "type": "@@search/AUTOCOMPLETE_FAILURE",
  },
]
`;

exports[`autocomplete actions autocomplete success Dispatches the autocomplete success action 1`] = `
Array [
  Object {
    "payload": Object {
      "isLoading": false,
      "suggestions": undefined,
    },
    "type": "@@search/AUTOCOMPLETE_SUCCESS",
  },
]
`;
